File: mycollab-web/src/main/scala/com/mycollab/module/file/events/FileEvent.scala
package com.mycollab.module.file.events

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.ecm.domain.Resource
import com.mycollab.module.ecm.domain.{ExternalDrive, Resource}

/**
  * @author MyCollab Ltd
  * @since 5.1.1
  */
object FileEvent {

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class ResourceRemovedEvent(source: AnyRef, data: Resource) extends ApplicationEvent(source, data) {}

  class ExternalDriveConnectedEvent(source: AnyRef, data: ExternalDrive) extends ApplicationEvent(source, data) {}

  class ExternalDriveDeleteEvent(source: AnyRef, data: ExternalDrive) extends ApplicationEvent(source, data) {}

}

File: mycollab-web/src/main/java/com/mycollab/module/file/event/FileEvent.kt
package com.mycollab.module.file.event

import com.mycollab.events.ApplicationEvent
import com.mycollab.module.ecm.domain.ExternalDrive
import com.mycollab.module.ecm.domain.Resource

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object FileEvent {
    class GotoList(source: Any, val data: Any?) : ApplicationEvent(source)

    class ResourceRemovedEvent(source: Any, data: Resource) : ApplicationEvent(source)

    class ExternalDriveConnectedEvent(source: Any, data: ExternalDrive) : ApplicationEvent(source)

    class ExternalDriveDeleteEvent(source: Any, data: ExternalDrive) : ApplicationEvent(source)
}